src/main/kotlin/io/github/nomisrev/routes/profile.kt

```diff
@@ -0,0 +1,42 @@
+package io.github.nomisrev.routes
+
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.raise.either
+import arrow.core.right
+import io.github.nomisrev.DomainError
+import io.github.nomisrev.MissingParameter
+import io.github.nomisrev.repo.UserPersistence
+import io.ktor.http.HttpStatusCode
+import io.ktor.resources.Resource
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.application.call
+import io.ktor.server.resources.get
+import io.ktor.server.routing.Route
+import io.ktor.util.pipeline.PipelineContext
+
+@Resource("/profiles")
+data class ProfileResource(val parent: RootResource = RootResource) {
+  @Resource("/{$USERNAME}")
+  data class Username(val parent: ProfileResource = ProfileResource(), val username: String)
+}
+
+fun Route.profileRoutes(
+  repo: UserPersistence,
+) {
+  get<ProfileResource.Username> {
+    either {
+        val username = parameter(USERNAME) { it }.bind()
+        repo.selectProfile(username).bind()
+      }
+      .respond(HttpStatusCode.OK)
+  }
```

Comment:
Type safe routing from Ktor gives us acces to an `ProfileResource.Username` instance, and thus safe acces to `username`.

Spotless also required a small formatting.

```suggestion
  get<ProfileResource.Username> { route ->
    either { repo.selectProfile(route.username).bind() }.respond(HttpStatusCode.OK)
  }
```

Reply:
Addressed in: [8c1113f](https://github.com/nomisRev/ktor-arrow-example/pull/186/commits/8c1113f2e4e27ecf34c454cff52821d8599dea35)